ksql-engine/src/main/java/io/confluent/ksql/planner/plan/AggregateNode.java

```diff
@@ -189,11 +193,18 @@ public SchemaKStream buildStream(
     // Pre aggregate computations
     final List<Pair<String, Expression>> aggArgExpansionList = new ArrayList<>();
     final Map<String, Integer> expressionNames = new HashMap<>();
-    collectAggregateArgExpressions(getRequiredColumnList(), aggArgExpansionList, expressionNames);
+    final Map<String, String> expressionToInternalColumnNameMap = new HashMap<>();
```

Comment:
Rather than passing around these lists and maps, maybe we can encapsulate them into an AggregateNode.InternalSchema class with accessors for fields by expression, internal name, and index? I think that would make the code here easier to understand.

Reply:
+1 passing around Maps, Lists, Sets etc etc does nothing for code readability, maintainability etc

Reply:
Going further on this. If we had an `AggregateNode.InternalSchema` type, then we could fold the `collectAggregateArgExpression` function calls into the constructor of this type. AT the top level, we would then just have 

`InternalSchema internalSchema = new InternalSchema(getRequiredColumnList(), getAggregateFunctionArguments())`;

Would be much more readable IMO.

Reply:
Good point, Using InternalSchema now!